ember-client/shared/src/main/scala/org/http4s/ember/client/internal/EmberWSClient.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.client.internal
+
+import cats.effect.Async
+import cats.effect.implicits._
+import cats.effect.std.Queue
+import cats.syntax.all._
+import org.http4s.Request
+import org.http4s._
+import org.http4s.client.Client
+import org.http4s.client.websocket.WSClient
+import org.http4s.client.websocket.WSConnection
+import org.http4s.client.websocket.WSFrame
+import org.http4s.ember.client.internal.WebSocketHelpers._
+import org.http4s.ember.core.WebSocketHelpers.decodeFrames
+import org.http4s.ember.core.WebSocketHelpers.frameToBytes
+import org.http4s.websocket.WebSocketFrame
+
+object EmberWSClient {
+  def apply[F[_]](
+      emberClient: Client[F]
+  )(implicit F: Async[F]): WSClient[F] =
+    WSClient[F](respondToPings = false) { wsRequest =>
+      val httpWSRequest = Request[F]()
+        .withUri(wsRequest.uri)
+        .withHeaders(wsRequest.headers)
+        .withMethod(Method.GET)
```

Comment:
I think there's a bug here: the `Uri` for `wsRequest` will be either `ws://` or `wss://` but this is not correct for a HTTP request. So we should rewrite that to `http://` or `https://` before making the request.